sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -1042,3 +1042,10 @@ def __sklearn_is_fitted__(self):
 
     check_fit_check_is_fitted("estimator", Estimator(behavior="method"))
     check_fit_check_is_fitted("estimator", Estimator(behavior="attribute"))
+
+
+# TODO: Remove in 1.3 when Estimator is deprecated
+def test_deprecated_Estimator_check_estimator():
+    with warnings.catch_warnings(record=True) as record:
+        check_estimator(Estimator=NuSVC())
+    assert FutureWarning in [rec.category for rec in record]
```

Comment:
Let's be extra strict here:

```python
    with warnings.catch_warnings(record=True) as record:
        check_estimator(Estimator=NuSVC())
    future_warnings = [
        rec
        for rec in record
        if issubclass(rec.category, FutureWarning)
        and str(rec.message) in "'Estimator' was deprecated in favor of"
    ]
    assert not future_warnings
```

This way we check specifically for the error raised by `check_estimator`.